pkg/kubeconfig/kubecfg_builder.go

```diff
@@ -186,7 +181,9 @@ func (b *KubeconfigBuilder) WriteKubecfg() error {
 		}
 
 		context.Cluster = b.Context
-		context.AuthInfo = b.Context
```

Comment:
I'm not quite sure what the reason for a `--user` flag is, but if a user specified `--admin` I'd think they'd want the `AuthInfo` set to the exported user.

Reply:
`--user` is to add an existing user to a new context. Since we use OIDC users, we just re-use the same user block for every cluster rather than specifying one user block per cluster.

If we allow both `--admin` and `--user` at the same time, and that this means "write the exported admin credentials to this user block" at least kops would not be able to delete that user when deleting the cluster. I am not sure what the use case would be for having a very specific name for the "kops managed" user block either. If we do want this feature, I'd rather have a new `--admin-user-name` flag or something.

Reply:
I would think that if just `--admin` was given then the expectation would be the current behavior of setting `AuthInfo` to the newly created context.

Reply:
Yes. That is the intention. The newly created context is always set to the cluster name, and here I set the user to be the same as the cluster name as well: https://github.com/kubernetes/kops/pull/9280/files#diff-2b7a768c601c36223ba9f3bafca41a31R137
So the effect should be the same.